### PR TITLE
Healthcheck for request URIs which are too long

### PR DESCRIPTION
**Pull request by gmethvin**

### CODE REVIEW

**Comment by 2is10** [comment]
Should we, perhaps, check the request URI length on the sending side rather than the receiving, in case it gets truncated somewhere along the way?

**Comment by gmethvin** [comment]
I think that might be a better approach. I'll add it to our ServiceClient instead of doing this.

File: kifi-backend/modules/common/app/com/keepit/common/service/ServiceClient.scala [hunk 1]
```diff
@@ -40,9 +45,16 @@ trait ServiceClient extends Logging {
 
   protected def call(call: ServiceRoute, body: JsValue = JsNull): Future[ClientResponse] = callUrl(call, url(call.url), body)
 
-  protected def callUrl(call: ServiceRoute, url: String, body: JsValue): Future[ClientResponse] = call match {
-    case c @ ServiceRoute(GET, _, _*) => httpClient.getFuture(url)
-    case c @ ServiceRoute(POST, _, _*) => httpClient.postFuture(url, body)
+  protected def callUrl(call: ServiceRoute, url: String, body: JsValue): Future[ClientResponse] = {
+    if (url.length > ServiceClient.MaxUrlLength) {
+      healthcheck.addError(HealthcheckError(callType = Healthcheck.INTERNAL, errorMessage = Some(ErrorMessage(
+        "Request URI too long!", Some(s"Request URI $url exceeds max length of ${ServiceClient.MaxUrlLength}"
```
Comment (2is10) [inline kifi-backend/modules/common/app/com/keepit/common/service/ServiceClient.scala]: Description might be a bit more readable like this: "Request URI length 1578 > 1000: http://...".

**Comment by 2is10** [comment]
Cool. Merge at will.

**Pull request merged**